gsy_e_sdk/setups/Asset_API_scripts/canary_network_live_data.py

```diff
@@ -0,0 +1,117 @@
+"""
+Template file to send live data to a Canary Network using Rest.
+"""
+import os
+from time import sleep
+from typing import List, Dict
+from pendulum import from_format
+from gsy_framework.constants_limits import DATE_TIME_FORMAT
+from gsy_e_sdk.aggregator import Aggregator
+from gsy_e_sdk.clients.rest_asset_client import RestAssetClient
+from gsy_e_sdk.utils import get_area_uuid_from_area_name_and_collaboration_id, get_assets_name
+
+ORACLE_NAME = "oracle"
+
+# List of assets's names to be connected with the API
+LOAD_NAMES = ["Load 1", "Load 2"]
+PV_NAMES = ["PV 1", "PV 2"]
+STORAGE_NAMES = []
+
+CONNECT_TO_ALL_ASSETS = False
+
+
+class Oracle(Aggregator):
+    """Class that defines the behaviour of an "oracle" aggregator."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.is_finished = False
+
+    def on_market_cycle(self, market_info):
+        """Place a bid or an offer whenever a new market is created."""
+        if self.is_finished is True:
+            return
+        self.send_forecasts(market_info)
+
+    def on_tick(self, tick_info):
+        pass
+
+    def send_forecasts(self, market_info):
+        """Send forecasts of the next market slot to the exchange."""
+        forecast_market_slot = (
+            from_format(market_info["market_slot"], DATE_TIME_FORMAT)
+            .add(minutes=15)
+            .format(DATE_TIME_FORMAT)
+        )
+        # pylint: disable=unused-variable
+        for area_uuid, area_dict in self.latest_grid_tree_flat.items():
+            asset_info = area_dict.get("asset_info")
+            if not asset_info:
+                continue
+
+            # Consumption assets
+            if "energy_requirement_kWh" in asset_info:
+                asset_name = area_dict["area_name"]
+                globals()[f"{asset_name}"].set_energy_forecast(
```

Comment:
I'm getting back to this, as using globals is usually discouraged. I assume you want to access the `RestAssetClient` instance for this specific asset. In theory, we should find a way to pass the list of assets to the Oracle (so that it can use it in this method, without accessing `globals()`). One way could be to add it to the `__init__()` (as I mentioned in the other PR) and probably refactor the `register_asset_list()` method.